polkadot/node/subsystem-bench/src/core/mock/network_bridge.rs

```diff
@@ -322,12 +133,55 @@ impl MockNetworkBridgeTx {
 	}
 }
 
-// A helper to determine the request payload size.
-fn request_size(request: &Requests) -> usize {
-	match request {
-		Requests::ChunkFetchingV1(outgoing_request) => outgoing_request.payload.encoded_size(),
-		Requests::AvailableDataFetchingV1(outgoing_request) =>
-			outgoing_request.payload.encoded_size(),
-		_ => unimplemented!("received an unexpected request"),
+#[overseer::contextbounds(NetworkBridgeRx, prefix = self::overseer)]
+impl MockNetworkBridgeRx {
+	async fn run<Context>(mut self, mut ctx: Context) {
+		// Main subsystem loop.
+		let mut from_network_interface = self.network_receiver.0;
+		loop {
+			futures::select! {
+				maybe_peer_message = from_network_interface.next() => {
+					if let Some(message) = maybe_peer_message {
+						match message {
+							NetworkMessage::MessageFromPeer(message) => match message {
+								Versioned::V2(
+									polkadot_node_network_protocol::v2::ValidationProtocol::BitfieldDistribution(
+										bitfield,
+									),
+								) => {
+									ctx.send_message(
+										BitfieldDistributionMessage::NetworkBridgeUpdate(NetworkBridgeEvent::PeerMessage(PeerId::random(), polkadot_node_network_protocol::Versioned::V2(bitfield)))
+									).await;
+								},
+								_ => {
+									unimplemented!("We only talk v2 network protocol")
+								},
+							},
+							NetworkMessage::RequestFromPeer(request) => {
+								if let Some(protocol) = self.chunk_request_sender.as_mut() {
+									if let Some(inbound_queue) = protocol.inbound_queue.as_ref() {
```

Comment:
longer term: we should map the request protocol to the inbound queue.

at least we should check that the protocol name matches and panic otherwise. when adding a new request to a benchmark, the subsystem will subtly fail to deserialize the request.